### PR TITLE
Combined permission request

### PR DESCRIPTION
**Pull request by toger5**
This is the part of the changes that need to be done to element-call. The sdk's need to be patched as well.
https://github.com/livekit/components-js/pull/533
https://github.com/livekit/client-sdk-js/pull/755

### CODE REVIEW

File: src/room/VideoPreview.tsx [hunk 1]
```diff
@@ -62,25 +61,32 @@ export function VideoPreview({ matrixInfo, onUserChoicesChanged }: Props) {
   }, [settingsModalState]);
 
   // Fetch user media devices.
-  const mediaDevices = useMediaDevices();
-
+  const mediaDevices = useMediaAllDevicesWithLocalStorage();
+  const { permittedDevices, videoIn, audioIn } = mediaDevices;
```
Comment (SimonBrandner) [inline src/room/VideoPreview.tsx:65]:
```suggestion
  const { permittedDevices, videoIn, audioIn } = useMediaAllDevicesWithLocalStorage();
```

**Review by SimonBrandner — APPROVED** [review]
Looks sane

File: src/room/VideoPreview.tsx (hunk 1 above)
Comment (toger5) [reply src/room/VideoPreview.tsx:65]: I wanted to shorten the return value that is why i am storing the full object in a seperate variable.

Comment (SimonBrandner) [reply src/room/VideoPreview.tsx:65]: Sorry, I am not sure I follow?

**Comment by toger5** [comment]
This will be closed in favor of: https://github.com/vector-im/element-call/pull/1200

**Pull request closed without merging**